src/Database/QueryCompiler.php

```diff
@@ -125,7 +125,9 @@ public function compile(Query $query, ValueBinder $generator)
     protected function _sqlCompiler(&$sql, $query, $generator)
     {
         return function ($parts, $name) use (&$sql, $query, $generator) {
-            if (!count($parts)) {
+            if (!isset($parts) ||
```

Comment:
Not for the full set of conditionals (if that was the question?).
Objects with Countable interfaces and count() == 0 are not considered "empty" by the empty function.

Also I am mimicing the old conditional's (possible) reliance on count returning 1 when an object was not an array or Countable.  In other words I _know_ that this new conditional acts exactly the same as `!count($parts)` but without the warning emitted.